File: Busqueda lineal 01-09-25/busqueda_lineal.cpp
#include <iostream>
using namespace std;

void imprimir(int arre[], int n) {
    cout << "El arreglo es: ";
    for (int i = 0; i < n; i++) {
        cout << arre[i] << " ";
    }
    cout << endl;
}

void insertar_elemento(int arre[], int n, int val, int indice) {
    for (int i = n - 1; i > indice; i--) {
        arre[i] = arre[i - 1];
    }
    arre[indice] = val;
}

int busqueda_lineal(int arre[], int n, int obj) {
    for (int i = 0; i < n; i++) {
        if (arre[i] == obj) {
            return i;
        }
    }
    return -1;
}

int main() {
    const int n = 5;
    int arre[n] = {1, 2, 3, 4, 5};
    imprimir(arre, n);

    int val, indice;
    cout << "Valor para insertar: ";
    cin >> val;
    cout << "Indice donde se va a insertar (0.." << n-1 << "): ";
    cin >> indice;

    insertar_elemento(arre, n, val, indice);
    imprimir(arre, n);

    int obj = val;
    int pos = busqueda_lineal(arre, n, obj);
    cout << "busqueda lineal de " << obj << ": indice " << pos << endl;

    cout << "\nPresiona ENTER para salir";
    cin.ignore();
    cin.get();
    return 0;
}

File: Busqueda lineal 01-09-25/busqueda_lineal.java
import java.util.Scanner;

public class Main {

    static void imprimir(int[] arre) {
        System.out.print("El arreglo es: ");
        for (int x : arre) {
            System.out.print(x + " ");
        }
        System.out.println();
    }

    static void insertar_elemento(int[] arre, int val, int indice) {
        for (int i = arre.length - 1; i > indice; i--) {
            arre[i] = arre[i - 1];
        }
        arre[indice] = val;
    }

    static int busqueda_lineal(int[] arre, int obj) {
        for (int i = 0; i < arre.length; i++) {
            if (arre[i] == obj) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int[] arre = {1, 2, 3, 4, 5};
        imprimir(arre);

        System.out.print("Valor para insertar: ");
        int val = sc.nextInt();

        System.out.print("Indice donde se va a insertar (0.." + (arre.length - 1) + "): ");
        int indice = sc.nextInt();

        insertar_elemento(arre, val, indice);
        imprimir(arre);

        int obj = val;
        int pos = busqueda_lineal(arre, obj);
        System.out.println("Busqueda lineal de " + obj + ": indice " + pos);

        System.out.println("\nPresiona enter para salir");
        try {
            System.in.read();
        } catch (Exception e) {}
    }
}
